asdf/schema.py

```diff
@@ -415,8 +406,7 @@ def load_schema(url, resolver=None, resolve_references=False):
         If ``True``, resolve all ``$ref`` references.
 
     """
-    if resolver is None:
-        resolver = _tag_to_uri
+    # TODO deprecate resolver?
```

Comment:
Part of me wonders if this is sufficient reason to delay 4.0 for another 3.x release.

`asdf.schema.load_schema` is a widely used part of the public API. However the `resolver` argument is I think unused (a more thorough look is needed). Keeping it significantly complicates this file. It was in place in part to allow `$ref` entries in schema files to resolve when provided `tag:foo:bar...` uris (which has long been deprecated and is being removed in asdf 4.0).

The other part of me knows that there is bound to be something we "miss" in the API cleanup and this so far is it.